### ISSUE TITLE
Elements are animating in before they're in view of the window

### ISSUE THREAD

**Issue by joshkarges** [michalsnik/aos#644]
## This is a:

- Bug

## Specifications

  - AOS version: 2.3.4
  - OS: Mac 10.15.7
  - Browser: Chrome 86.0.4240.198

## Expected Behavior

When each element with the `data-aos="fade-up"` attribute scrolls into view, it will animate into visibility.

## Actual Behavior

The elements are animating in _before_ they are scrolled into view.  The first element appears when the top is about 120px above the bottom of the window, then the second appears when its top is about 10px above the bottom of the window, then the subsequent elements appear when their top is even below the bottom of the window.

## Steps to Reproduce the Problem

Here's my page:
```javascript
const useStyles = makeStyles((theme) => ({
  contentContainer: {
    position: 'relative',
    paddingBottom: '56.25%',
    height: 0,
  },
  video: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%',
    zIndex: -1,
  },
}));

const HomePage = () => {
  const classes = useStyles();
  return (
    <FlexBox flexDirection='column'>
      <FlexBox className={classes.contentContainer}>
        <video muted id="background-video" loop autoPlay className={classes.video}>
          <source src={videoSrc} type="video/mp4" />
          <source src={videoSrc} type="video/ogg" />
          Your browser does not support the video tag.
        </video>
      </FlexBox>
      <div data-aos="fade-up"><ServicesPage /></div>
      <div data-aos="fade-up"><PortfolioPage /></div>
      <div data-aos="fade-up"><AboutPage /></div>
      <div data-aos="fade-up"><ContactPage /></div>
    </FlexBox>
  );
};
```

_UPDATE1:_
After a bit more investigation, I think it may have to do with when AOS is `init`ed.  Right now, I just call `Aos.init()` at the global level (not in a component function), in my App.tsx file.  I think I should maybe call it in a `useEffect` or `useLayoutEffect`, since my pages have some hefty images they need to load/render.  Anyone have any tips on this?

_UPDATE2:_
Confirmed that my `AOS.init()` timing is the issue.  When I resize the window, and return it to the same size, it works as expected.  Setting it in a `useEffect` or `useLayoutEffect` doesn't solve the issue yet, since my "PortfolioPage" has an offsetHeight of 327 at the time of those hooks, but eventually has an offsetHeight of 1598.

**Comment by Lordor77** [comment]
Hey, I'm having the same issue. Any solutions?

**Comment by joshkarges** [comment]
> Hey, I'm having the same issue. Any solutions?

Not really.  It was my "Services Page" that was getting calculated with the wrong height by Aos.  I eventually just tweaked how the Services Page was laid out, and it eventually started working, but I can't point to one specific thing that was the cause.
